app/src/test/java/msi/paria/trainingapp/SampleViewModelTest.kt

```diff
@@ -0,0 +1,81 @@
+package msi.paria.trainingapp
+
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class SampleViewModelTest {
+    lateinit var sampleViewModel: SampleViewModel
+
+    @Before
+    fun initValues() {
+        sampleViewModel = SampleViewModel()
+    }
+
+    @Test
+    fun GivenCorrectEmail_WhenEmailChangedAndSubmitClicked_ThenValidationShouldBeTrue() {
+        val email = "paria.m7616@gmial.com"
+        sampleViewModel.onEmailChanged(email)
+        sampleViewModel.onSubmitButtonClicked()
+        assertTrue(sampleViewModel.emailValidateState.value)
+    }
+
+    @Test
+    fun GivenWrongEmail_WhenEmailChangedAndSubmitClicked_ThenValidationShouldBeFalse() {
+        val email = "paria."
+        sampleViewModel.onEmailChanged(email)
+        sampleViewModel.onSubmitButtonClicked()
+        assertFalse(sampleViewModel.emailValidateState.value)
+    }
+
+    @Test
+    fun GivenEmptyEmail_WhenEmailChangedAndSubmitClicked_ThenValidationShouldBeFalse() {
+        val email = ""
+        sampleViewModel.onEmailChanged(email)
+        sampleViewModel.onSubmitButtonClicked()
+        assertFalse(sampleViewModel.emailValidateState.value)
+    }
+
+    @Test
+    fun GivenCorrectEmail_WhenEmailChangedAndSubmitNotClicked_ThenValidationShouldBeFalse() {
+        val email = "paria.m7616@gmial.com"
+        sampleViewModel.onEmailChanged(email)
+        assertFalse(sampleViewModel.emailValidateState.value)
+    }
+
+    @Test
+    fun GivenCorrectEmail_WhenEmailNotChangedAndSubmitClicked_ThenValidationShouldBeFalse() {
+        val email = "paria.m7616@gmial.com"
+        sampleViewModel.onSubmitButtonClicked()
+        assertFalse(sampleViewModel.emailValidateState.value)
+    }
+    @Test
+    fun GivenMultipleEmail_WhenEmailChangedAndSubmitClicked_ThenValidationShouldBeFalse() {
+        var email = "paria.m7616@gmial.com"
+        email = "pariamsi.m7616@gmial.com"
```

Comment:
You are not reacting with the view model here. I mean if you change the email locally and do not inform the view model about this change, it doesn't make sense.